src/iamserver/nodeinfoprovider.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <filesystem>
+#include <fstream>
+
+#include "log.hpp"
+#include "nodeinfoprovider.hpp"
+#include "utils/exception.hpp"
+#include "utils/systeminfo.hpp"
+
+/***********************************************************************************************************************
+ * Constants
+ **********************************************************************************************************************/
+
+constexpr auto cProvisioningStatusPath = "/var/aos/.provisionstate";
```

Comment:
Add word `default`. Open question where should we handle default cases? In config or in client module?